core/sources/domains.move

```diff
@@ -135,7 +157,19 @@ module aptos_names::domains {
         token_helper::initialize(account);
     }
 
-    fun register_domain_generic(sign: &signer, domain_name: String, num_years: u8) acquires NameRegistryV1, RegisterNameEventsV1, SetNameAddressEventsV1 {
+    public fun init_reverse_lookup_registry_v1(account: &signer) {
```

Comment:
Not sure if I missed this in the initial review for testnet but can we change this to public entry fun? There's no reason for this to be non-entry and having to execute this using the script (sources/script.move). Changing from public fun to public entry fun is a compatible upgrade, so we should likely make a similar change in the testnet branch and upgrade the testnet contract.

Reply:
but to be safe should we add a check in `init_reverse_lookup_registry_v1` to make sure only creating the revers lookup table when there isn't one?

Reply:
Technically you don't need to. move_to will fail if you're creating the same resource twice for the same account. The error message is not as good as an explicit assertion but in some cases like this where a function is only meant to be called once, we don't need nice error messages.